geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/RowSkippingIterator.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.core.iterators
+
+import java.util.{Collection => jCollection}
+
+import org.apache.accumulo.core.client.IteratorSetting
+import org.apache.accumulo.core.data.{Range => AccRange, _}
+import org.apache.accumulo.core.iterators.{IteratorEnvironment, SortedKeyValueIterator}
+import org.apache.hadoop.io.Text
+import org.locationtech.geomesa.core.iterators.RowSkippingIterator._
+
+import scala.collection.SortedSet
+
+/**
+ * Iterator that will skip rows based on a set of suffix strings.
+ */
+class RowSkippingIterator extends GeomesaFilteringIterator {
```

Comment:
The original idea was to use less ranges and then skip over rows we know aren't part of our query area. But in practice it wasn't any faster in my (limited) testing, so we decided to hold off. It probably warrants further testing in geobench or the stuff @jahhulbert-ccri is working on.